src/LittleForker/CooperativeShutdown.cs

```diff
@@ -30,15 +28,19 @@ public static class CooperativeShutdown
         ///     The callback that is invoked when cooperative shutdown has been
         ///     requested.
         /// </param>
+        /// <param name="loggerFactory">
+        ///     A logger factory.
+        /// </param>
         /// <param name="onError">A method to be called if an error occurs while listening</param>
         /// <returns>
         ///     A disposable representing the named pipe listener.
         /// </returns>
-        public static Task<IDisposable> Listen(Action shutdownRequested, Action<Exception> onError = default)
+        public static Task<IDisposable> Listen(Action shutdownRequested, ILoggerFactory loggerFactory, Action<Exception> onError = default)
```

Comment:
Logging might be optional maybe?
`loggerFactory ??= NullLoggerFactory.Instance;`

Reply:
I'm ok either way. Do you specifically want this?